Add notable strat in Red Tower for R-mode frozen beetom x-ray climb

blkerby opened the pull request:
Video by Bobbob: https://vimeo.com/810399620

Oatsngoats did this on stream last night and got it on his first attempt.

osse101 reviewed (commented):
A `canRModeForceStand` is reasonable to package together with the `canEnterRMode` tech.

This needs to indicate you leave the room with `Reserve Health minus a Beetom hit and some Ripper hits.`  
The note should clarify where you stand before jumping with the beetom because it is pixel perfect on X.
Bob indicated in the video description there is a way to do the strat by getting the beetom stuck in the wall and walk from the frozen ripper.  Something to test and verify.

blkerby commented:
Yeah I think forced stand can be assumed as part of canEnterRMode (similar to how with G-mode we assume they can overload PLMs, rather than having a separate tech).

I added enemyDamage requirements to give the correct final health. The comeInWithRMode includes reserve energy being transferred to regular energy, and since since the farm is freely available and can be used before doing the strat, reserves can be full at the start. So this makes it unnecessary to explicitly model the farm in the middle of the strat.

I added a clarification about the X position in the note.

I tried that other strat from the video description but wasn't able to get it to work (probably I've misunderstood part of it). Maybe we can just add this one for now?

osse101 commented:
Yep, it would just be an alternate way to do the same thing.
